x-pack/plugins/observability/public/plugin.ts

```diff
@@ -401,6 +401,19 @@ export class Plugin
               ]
             : [];
 
+          const aiAssistantLink =
+            !Boolean(pluginsSetup.serverless) && Boolean(pluginsSetup.observabilityAIAssistant)
```

Comment:
There might be other conditions we want to check also (eg license). 
Afaict we don't check for license in other places before showing links to the assistant 

https://github.com/elastic/kibana/blob/f74c0e17aad64fc359da6bf3fd7255ced8cfb146/x-pack/plugins/apm/public/components/routing/app_root/apm_header_action_menu/index.tsx#L104-L106

Reply:
@dgieselaar any insight?

Reply:
This has come up before, and generally we have depended on the AI connector to serve as a license check since you can't create one without the right license. 

In this case it makes sense to add a license check so we don't show the nav item for those that don't have the right license.

Reply:
I think we should use the same logic as we do for showing the AI Assistant button, which I think is the case here - but it might make sense to abstract that into a specific function, like `shouldShowAssistantLinks` or something  similar.

Reply:
Actually, we should use `service.isEnabled()` - that one uses the Kibana feature flag.

Reply:
@dgieselaar Afaict `isEnabled` is simply:

```
const isEnabled = coreStart.application.capabilities.observabilityAIAssistant.show;
```

https://github.com/elastic/kibana/blob/d84ac07b2e82714d88eb17dc5419d74971e09606/x-pack/plugins/observability_ai_assistant/public/plugin.tsx#L107

`service.isEnabled` is only available inside the AI Assistant plugin. You okay with me just checking for `coreStart.application.capabilities.observabilityAIAssistant.show`?

Btw this means that we show the AI assistant link regardless of license, right?